README.md

```diff
@@ -1 +1,80 @@
+
+
+## Setup
+
+To prevent confusion, in previous lessons I manually turned off the strong parameter requirement, to start this lesson I've gone with the Rails 4+ default that requires you to use strong parameters. If for any reason you ever need to turn them off you can add the line `config.action_controller.permit_all_parameters = true` to the `config/application.rb` file.
+
+
+## Code Implementation
+
+Let's begin by running the rails server and navigating to `localhost:3000/posts/new`, once there fill out the form and click `submit`. You'll see we get the following `ForbiddenAttributesError`:
+
+![ForbiddenAttributesError](https://s3.amazonaws.com/flatiron-bucket/readme-lessons/ForbiddenAttributesError.png)
+
+What this means is that Rails needs to be told what parameters are allowed to be submitted through the form to the database. Remembering back to our pharmacy example, this is essentially like a pharmacist not giving out a prescription out unless they have:
+
+1. A valid prescription from a doctor
+
+2. The customer requesting the item matches the name on the prescription
+
+The same error would occur if you were trying to update a record, so how do we fix this? Let's update the `create` and `update` methods to look like the code below:
+
+```ruby
+# app/controllers/posts_controller.rb
+
+def create
+  @post = Post.new(params.require(:post).permit(:title, :description))
+  @post.save
+  redirect_to post_path(@post)
+end
+
+def update
+  @post = Post.find(params[:id])
+  @post.update(params.require(:post).permit(:title, :description))
+  redirect_to post_path(@post)
+end
+```
+
+If you go back to the web browser and click refresh you'll see everything is working for both the `create` and `update` actions. Running the Rspec tests reveals that our specs are now passing again as well.
+
+
+## DRYing up Strong Params
+
+The code we wrote above is great if you only have a `create` method in your controller, however if you have a standard CRUD setup you will also need to implement the same code in your `update` action. It's a standard Rails practice to remove code repetition, so let's abstract the strong parameter call into its own method in the controller:
+
+```ruby
+# app/controllers/posts_controller.rb
+
+private
+
+  def post_params
+    params.require(:post).permit(:title, :description)
+  end
+```
+
+Now, both our `create` and `update` methods in the `posts` controller can simply call `post_params`.
+
+```ruby
+# app/controllers/posts_controller.rb
+
+def create
```

Comment:
Can you include the method definition again in this code sample. I was a bit confused where the method went